src/data_generators/data_set_parameters.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines the parameters for generating a DataDesign."""
+
+from numpy.random import Generator
+from typing import NamedTuple
+from typing import Type
+from typing import Any
+
+
+class GeneratorParameters(NamedTuple):
+  """Parameters to create one PricingGenerator.
+
+    generator: Class of the generator (e.g. HomogeneousImpressionGenerator)
+    params : Parameters dict used to initialize that class.
+    """
+  #  TODO(uakyol): create a Generator parent class for impression, pricing and
+  #  overlap generators and change this type.
+  generator: Type[Any]
+  params: dict
+
+
+class DataSetParameters(NamedTuple):
+  """Parameters to create one DataSet.
+
+  Does not include randomness
+
+    num_reps:  Number of repetitions for this DataSet to be randomly generated.
+    num_publishers: Number of publishers in this DataSet.
+    largest_publisher_size: Maximum possible reach of the largest publisher.
+    publisher_size_decay_rate: Decay rate for calculating publisher sizes from
+        the largest publisher size.
+    pricing_generator_params: Parameters for the pricing generator.
+    impression_generator_params: Parameters for the impression generator.
+    overlap_generator_params: Parameters for the overlap generator.
+    name: Filename of the data set to be generated when written to disk.
```

Comment:
Can this be generated on-the-fly from the parameters in this structure?  Ideally, it would be nice to have a name that is at least partially human readable, so that we easily investigate the behavior of a specific data set if there is a need to do that.

Reply:
yes, it can be. I thought about that but the problem is there are so many parameters that the name would become meaningless if I just concatenate their names and values. One option is to not include the sub parameters (parameters defined in the dict of generators) and just concatenate the others with their values. This can be implemented as a parent function in SyntheticDataDesignConfig and if needed be overwritten in child configs. wdyt?

Reply:
I think that's fine.  You might want to append this with a randomly generated string so that there aren't any name collisions.  It's a requirement that the names be unique.

Reply:
sg, changed it.